nvim/util.lua

```diff
@@ -0,0 +1,218 @@
+------------
+-- util
+-- utility functions
+-- module: util
+
+local Util = {}
```

Comment:
Seems like `nvim.table` is more appropriate name for this module than `nvim.util`. Until now we avoided a "util" or "helper" module, we should try to maintain that.